File: problems/1-easy/1-Two-Sum/solution.cpp
#include <vector>
#include <unordered_map>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> my_map;
    for (int i = 0; i < nums.size(); i++) {
        int diff = target - nums[i];
        if (my_map.find(diff) != my_map.end()) {
            return {my_map[diff], i};
        } 
        my_map[nums[i]] = i;
    }
    return {};
}

File: problems/1-easy/1-Two-Sum/solution.java
import java.util.HashMap;
import java.util.Map;

public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> myMap = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
        int diff = target - nums[i];
        if (myMap.containsKey(diff)) {
            return new int[]{myMap.get(diff), i};
        }
        myMap.put(nums[i], i);
    }
    return new int[]{};
}